src/fortran_routing/mc_pylink_v00/MC_singleSeg_singleTS/test_MC_kernel.py

```diff
@@ -0,0 +1,52 @@
+import pytest
+import mc_sseg_stime_NOLOOP_demo as demo
+from test_suite_parameters import generate_conus_MC_parameters
+
+
+input_tup = (
```

Comment:
These tests should be restructured.

Consider:
```python
import pytest
import mc_sseg_stime_NOLOOP_demo as demo
from test_suite_parameters import generate_conus_MC_parameters

@pytest.fixture
def random_input_tuple():
	return generate_conus_MC_parameters(1, 16)

def test_MC_kernel(random_input_tuple):
	out = demo.compare_methods("single", *(next(params)))
	# print(out)
	qdc1, qdc2, velc1, velc2, depthc1, depthc2 = out
	assert (qdc1, velc1, depthc1) == (qdc2, velc2, depthc2)
```

Reply:
You might consider using hypothesis for generating test data.  https://hypothesis.readthedocs.io/en/latest/index.html

Reply:
@groutr: Is there a way to have that test result show the output for each test individually? It is very helpful (and potentially diagnostic) to know that the flow matches while the velocity and depth do not, for instance.

Reply:
Answering my own question --- `pytest -v`...

Reply:
That's probably the best way. With pytest, I really try to avoid relying on a global state as it's not certain to always be consistent.

Reply:
Another strategy is to use an addon for pytest, pytest-check (https://github.com/okken/pytest-check)